p2p/simulations/http.go

```diff
@@ -410,9 +412,38 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	// check if filtering has been requested
+	if fp := req.URL.Query().Get("filterProtocol"); fp != "" {
```

Comment:
that might be possible and desirable indeed. just keep in mind that from a frontend point of view, the more messages are enabled, the more it is possible that the visualisation grinds to a halt or becomes unusable due to message overload